### PR TITLE
docs: add example for :awscc_amplify_domain

### PR DESCRIPTION
**Pull request by novekm**
### Community Note

* Please vote on this pull request by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original pull request comment to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for pull request followers and do not help prioritize the request
* The resources and data sources in this provider are generated from the CloudFormation schema, so they can only support the actions that the underlying schema supports. For this reason submitted bugs should be limited to defects in the generation and runtime code of the provider. Customizing behavior of the resource, or noting a gap in behavior are not valid bugs and should be submitted as enhancements to AWS via the CloudFormation Open Coverage Roadmap.

Relates OR Closes #0000

### CODE REVIEW

File: examples/resources/awscc_amplify_domain/amplify_domain.tf [hunk 1]
```diff
@@ -0,0 +1,35 @@
+resource "awscc_amplify_app" "example" {
+  name = "app"
+
+  # Setup redirect from https://example.com to https://www.example.com
+  custom_rules = [
```
Comment (jackywong-amazon) [inline examples/resources/awscc_amplify_domain/amplify_domain.tf:5]: Suggestion here based on personal experience, I would rather to have another example for redirecting from apex domain to sub domain like www.example.com

Some customers required to use their apex domain as their company default domain.
For the basic amplify_domain, probably you can use a custom_rules for redirecting to a HTTP status code 404 page.

File: examples/resources/awscc_amplify_domain/amplify_domain.tf [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+}
+
+resource "awscc_amplify_branch" "main" {
+  app_id      = awscc_amplify_app.example.app_id
+  branch_name = "main"
+}
+
+resource "awscc_amplify_domain" "example" {
+  app_id      = awscc_amplify_app.example.app_id
+  domain_name = "example.com"
```
Comment (jackywong-amazon) [inline examples/resources/awscc_amplify_domain/amplify_domain.tf:21]: Quick question, would it be any use case or example using Route53 CNAME creation?

Comment (novekm) [reply examples/resources/awscc_amplify_domain/amplify_domain.tf:21]: Hmm, potentially but may get in the weeds of having to explain CNAME validation. I pulled and refactored this example from the docs for the `aws` provider for this resource. I'm thinking they may have left it out for similar reasons: https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/amplify_domain_association

Comment (novekm) [reply examples/resources/awscc_amplify_domain/amplify_domain.tf:21]: I could also potentially add something similar to this and then reference the domain name:
`resource "aws_route53_zone" "primary" {
  name = "example.com"
}`
but there are other resources that would also need to be added

**Review by jackywong-amazon — APPROVED** [review]
LGTM, thanks for your contribution!

**Comment by marcosentino** [comment]
@novekm Thank you for the contribution!

**Pull request merged**